### PR TITLE
Updated physical quantity docs for 3.1.5 API changes

### PR DESCRIPTION
**Pull request by dlewissandy**

### CODE REVIEW

File: services/scientist/maana-physical-quantity.md [hunk 1]
```diff
@@ -432,7 +728,7 @@ schema { query: Query }
 | Example | How can I see all the available physical dimensions? |
 | :--- | :--- |
 | Info | Since the system can derive an infinite number of physical dimensions using multiplication, division and exponentiation, It is NOT possible to see ALL available physical dimensions. Instead, we can use the physicalDimensions query to find all the physical dimensions that have been explicitly defined. The physical quantity service comes out of the box with over 60 of the most commonly occurring physical dimensions in engineering and scientific literature. |
-| Query | { query { physicalDimensions {  name}}} |
+| Query | query { physicalDimensions {  name alias formula}} |
```
Comment (aelkholy) [inline services/scientist/maana-physical-quantity.md]: typo 'alias' should be 'aliases'

File: services/scientist/maana-physical-quantity.md [hunk 2]
```diff
@@ -444,7 +740,7 @@ schema { query: Query }
 | Example | How do I get the preferred name of a physical dimension? |
 | :--- | :--- |
 | Info | Use the name field of the PhysicalDimension type to get the preferred name for a physical dimension. The example below gets the preferred name of all the explicitly defined physical dimensions: |
-| Query | { query { physicalDimension\(dimensionRef: {  name}}} |
+| Query | query { physicalDimension\(dimensionRef: {  name}} |
```
Comment (aelkholy) [inline services/scientist/maana-physical-quantity.md]: This query is mangled

File: services/scientist/maana-physical-quantity.md [hunk 3]
```diff
@@ -467,22 +763,10 @@ schema { query: Query }
 | Query | query {  systemsOfUnits{  name, description}} |
```
Comment (aelkholy) [inline services/scientist/maana-physical-quantity.md]: output does not match this

File: services/scientist/maana-physical-quantity.md [hunk 4]
```diff
@@ -502,56 +786,55 @@ schema { query: Query }
...
 | :--- | :--- |
-| Info | Use the optional systemRefs parameter to specify either the id or name of the required systems of units. If no systemRefs parameter is provided, the extractor uses all available systemsOfUnits. |
-| Query | { extract\(sources: \[ "the airspeed of the plane was 350 mph as it passed through 3000 meters.", "The minimum runway length is 7500 feet."\], systemRefs:\[{name:"Imperial"}\]\) {  entity {  name}}} |
-| Response | { "data": { "extract": \[ { "entity": { "name": "350.0 mi/hr"}}, { "entity": { "name": "7500.0 ft"}}\]}} |
+| Info | Use the optional systemInputs parameter to specify either the id or name of the required systems of units. If no systemInputs parameter is provided, the extractor uses all available systemsOfUnits. |
+| Query | query IMPERIAL{systemOfUnits\(systemRef:{name:"Imperial"}\){ ... SystemOfUnitsFragment}}<br>query EXAMPLE\($IMPERIAL:SystemOfUnitsInput\){ extractBatch\(sources: \["the airspeed of the plane was 350 mph as it passed through 3000 meters.","The minimum runway length is 7500 feet."\], systemInputs:\[$IMPERIAL\]\) {  entity {  name}}} |
```
Comment (aelkholy) [inline services/scientist/maana-physical-quantity.md]: does not work

File: services/scientist/maana-physical-quantity.md [hunk 5]
```diff
@@ -502,56 +786,55 @@ schema { query: Query }
...
 | Response | { "data": { "parse": { "name": "400.0 lb/in²"}}} |
 
 ### Extraction
 
 | Example | You have a list of strings, each which may contain zero or many physical quantities. How do you extract all the physical quantities from the source text? |
 | :--- | :--- |
 | Info | Use the extract query to get all the entity mentions from the text. The extract method can be used to find entity mentions in a list of string literals, or the field of a kind. The extractor will find all the physical quantity mentions in each string \(or instance of a field\) and return a list of the results. |
-| Query | { extract\(sources: \[ "the airspeed of the plane was 350 mph as it passed through 3000 meters.", "The minimum runway length is 7500 feet."\]\) {  entity {  name}}} |
-| Result | { "data": { "extract": \[ { "entity": { "name": "350.0 mi/hr"}}, { "entity": { "name": "3000 m"}}, { "entity": { "name": "7500.0 ft"}}\]}} |
+| Query | extractBatch\(sources: \[ "the airspeed of the plane was 350 mph as it passed through 3000 meters.", "The minimum runway length is 7500 feet."\]\) {  entity {  name}} |
```
Comment (aelkholy) [inline services/scientist/maana-physical-quantity.md]: query is malformed  (missing 'query')

File: services/scientist/maana-physical-quantity.md [hunk 6]
```diff
@@ -502,56 +786,55 @@ schema { query: Query }
...
-| Query | { extract\(sources: \[ "the airspeed of the plane was 350 mph as it passed through 3000 meters.", "The minimum runway length is 7500 feet."\], dimensionRef:{name:"LENGTH"}\) {  entity {  name} fromText, fromSpan, fromOffset}} |
-| Response | { "data": { "extract": \[ { "entity": { "name": "3000.0 m"}, "fromText": "the airspeed of the plane was 350 mph as it passed through 3000 meters.", "fromSpan": "12", "fromOffset": "59"}, { "entity": { "name": "7500.0 ft"}, "fromText": "The minimum runway length is 7500 feet.", "fromSpan": "10", "fromOffset": "29"}\]}} |
+| Query | query LENGTH{physicalDimension\(dimensionRef:{name:"PRESSURE"}\){ ... PhysicalDimensionFragment }}<br>query EXAMPLE($PRESSURE:PhysicalDimensionInput){ extractBatch\(sources: \[ "the airspeed of the plane was 350 mph as it passed through 3000 meters.", "The minimum runway length is 7500 feet."\], dimensionInputs:\[$LENGTH\] \) {  entity {  name} fromText, fromSpan, fromOffset}} |
```
Comment (aelkholy) [inline services/scientist/maana-physical-quantity.md]: I STILL CAN'T GET THIS TO WORK EVEN AFTER READING YOUR DUMB INSTRUCTIONS!!!

File: services/scientist/maana-physical-quantity.md [hunk 7]
```diff
@@ -564,14 +847,55 @@ schema { query: Query }
...
+| Response | { "data": { "product": { "name": "4.0 Hz·acre·in", "unitOfMeasure": { "dimension": { "formula": "L³T⁻¹"}}}}} |
 
 ### Size Comparisons
 
 | Example | Comparing size without converting units |
 | :--- | :--- |
 | Info | The physical quantity service will perform the necessary unit conversions for you, as long as the units are dimensionally consistent |
-| Query | { extract\(sources: \["1 inch","1 foot","1 yard"\]\) {  entity{  isGT\(rhs:{parseFrom:"2 1/2 feet"}\), isGTE\(rhs:{parseFrom:"2.54 centimeters"}\), isEqual\(rhs:{parseFrom:"3 feet"}\), isLT\(rhs:{parseFrom:"0.5 meters"}\), isLTE\(rhs:{parseFrom:"1/5280 mile"}\)}}} |
-| Response | { "data": { "extract": \[ { "entity": { "isGT": false, "isGTE": true, "isEqual": false, "isLT": true, "isLTE": true}}, { "entity": { "isGT": false, "isGTE": true, "isEqual": false, "isLT": true, "isLTE": true}}, { "entity": { "isGT": true, "isGTE": true, "isEqual": true, "isLT": false, "isLTE": false}}\]}} |
+| Query | query REFERENCE{ parse(source:"35 cm") { ... PhysicalQuantityFragment }}<br>query EXAMPLE($REF:PhysicalQuantityInput!){extractBatch(sources: ["1 inch","1 foot","1 yard"]) { entity{  isGT(rhs:$REFERENCE)}}} |
```
Comment (aelkholy) [inline services/scientist/maana-physical-quantity.md]: Used REF in variable def, then REFERENCE in query

**Review by aelkholy — COMMENTED** [review]
"Example    How can I see all the available unit of measure prefixes?"
and
"Example    How do I get the preferred symbol for a unit of measure prefix?"
have the same query.

`"data": {
    "unitOfMeasurePrefixes": [
      {
        "aliases": [
          "f",
          "fempto"
        ]
      },`
"fempto"

multiplier T is duplicated: 
`{
        "name": "T",
        "multiplier": 1000000000000
      }
    ]`

`query { physicalDimensions { name aliases formula}}` has no aliases for any of them

`{ physicalDimension(dimensionRef:{formula:"L²M¹T⁻²"}){ aliases}}` output does not match quide output.

`query { systemsOfUnits{ name, description}}` does not match guide output

Neither 'How do I convert the string to a physical quantity only if it has a specific physical dimension?' or 'How do I convert the string to a physical quantity only if its unit of measure is in a particular system of units?' work.

File: services/scientist/maana-physical-quantity.md (hunk 2 above)
Comment (dlewissandy) [reply services/scientist/maana-physical-quantity.md]: Thanks.  Fixed it.

File: services/scientist/maana-physical-quantity.md (hunk 4 above)
Comment (dlewissandy) [reply services/scientist/maana-physical-quantity.md]: Thanks.   Fixed it.

File: services/scientist/maana-physical-quantity.md (hunk 6 above)
Comment (dlewissandy) [reply services/scientist/maana-physical-quantity.md]: Looks like there is a typo in the query.   The variable named $PRESSURE is not defined.

File: services/scientist/maana-physical-quantity.md (hunk 7 above)
Comment (dlewissandy) [reply services/scientist/maana-physical-quantity.md]: Good get.   Fixed it.

File: services/scientist/maana-physical-quantity.md (hunk 5 above)
Comment (dlewissandy) [reply services/scientist/maana-physical-quantity.md]: The query keyword is optional.   For consistency I have added it back in everywhere.

File: services/scientist/maana-physical-quantity.md (hunk 3 above)
Comment (dlewissandy) [reply services/scientist/maana-physical-quantity.md]: Good catch! Updated results

**Comment by dlewissandy** [comment]
> "Example How can I see all the available unit of measure prefixes?"
> and
> "Example How do I get the preferred symbol for a unit of measure prefix?"
> have the same query.
> 

yes they do.   This is by design.   Since the documentation as arranged as a bunch of how-to's there is some redundancy.

**Comment by dlewissandy** [comment]
> "data": { "unitOfMeasurePrefixes": [ { "aliases": [ "f", "fempto" ] },
> "fempto"

Fempto is a valid alias.  https://en.wikipedia.org/wiki/Femto-

**Comment by dlewissandy** [comment]
> 
> multiplier T is duplicated:
> { "name": "T", "multiplier": 1000000000000 } ]

Looks like this is an error in the server.   I have filed a bugfix ticket.   This will be resolved prior to the release of version 3.1.5

**Comment by dlewissandy** [comment]
> query { physicalDimensions { name aliases formula}} has no aliases for any of them
> { physicalDimension(dimensionRef:{formula:"L²M¹T⁻²"}){ aliases}} output does not match quide output.
> 

This is a defect in the server.   I have filed a bugfix ticket.   This will be resolved prior to the release of 3.1.5

**Comment by dlewissandy** [comment]
> query { systemsOfUnits{ name, description}} does not match guide output

I have updated the documentation to reflect the correct output

**Comment by dlewissandy** [comment]
> Neither 'How do I convert the string to a physical quantity only if it has a specific physical dimension?' or 'How do I convert the string to a physical quantity only if its unit of measure is in a particular system of units?' work.

I have revised the queries.  Both contained typos.

**Pull request merged**